chips/nrf51/src/radio.rs

```diff
@@ -0,0 +1,456 @@
+//! Radio/BLE Driver for nrf51dk
+//!
+//! Sending BLE advertisement packets
+//! Possible payload is 30 bytes
+//!
+//! Author: Niklas Adolfsson <niklasadolfsson1@gmail.com>
+//! Author: Fredrik Nilsson <frednils@student.chalmers.se>
+//! Date: June 3, 2017
+
+use chip;
+use core::cell::Cell;
+use kernel::hil::time::Client;
+use kernel::returncode::ReturnCode;
+use nvic;
+use peripheral_interrupts::NvicIdx;
+use peripheral_registers::{RADIO_REGS, RADIO_BASE};
+
+
+pub const PACKET0_S1_SIZE: u32 = 0;
+pub const PACKET0_S0_SIZE: u32 = 0;
+
+pub const RADIO_PCNF0_S0LEN_POS: u32 = 8;
+pub const RADIO_PCNF0_S1LEN_POS: u32 = 16;
+pub const RADIO_PCNF0_LFLEN_POS: u32 = 0;
+
+pub const RADIO_PCNF1_WHITEEN_DISABLED: u32 = 0;
+pub const RADIO_PCNF1_WHITEEN_ENABLED: u32 = 1;
+pub const RADIO_PCNF1_WHITEEN_POS: u32 = 25;
+
+pub const RADIO_PCNF1_BALEN_POS: u32 = 16;
+pub const RADIO_PCNF1_STATLEN_POS: u32 = 8;
+pub const RADIO_PCNF1_MAXLEN_POS: u32 = 0;
+
+pub const RADIO_PCNF1_ENDIAN_POS: u32 = 24;
+pub const RADIO_PCNF1_ENDIAN_BIG: u32 = 1;
+
+pub const PACKET_LENGTH_FIELD_SIZE: u32 = 0;
+pub const PACKET_PAYLOAD_MAXSIZE: u32 = 64;
+pub const PACKET_BASE_ADDRESS_LENGTH: u32 = 4;
+pub const PACKET_STATIC_LENGTH: u32 = 64;
+
+
+// internal Radio State
+pub const RADIO_STATE_DISABLE: u32 = 0;
+pub const RADIO_STATE_RXRU: u32 = 1;
+pub const RADIO_STATE_RXIDLE: u32 = 2;
+pub const RADIO_STATE_RX: u32 = 3;
+pub const RADIO_STATE_RXDISABLE: u32 = 4;
+pub const RADIO_STATE_TXRU: u32 = 9;
+pub const RADIO_STATE_TXIDLE: u32 = 10;
+pub const RADIO_STATE_TX: u32 = 11;
+pub const RADIO_STATE_TXDISABLE: u32 = 12;
+
+
+// static mut TX_BUF: [u8; 16] = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
+static mut RX_BUF: [u8; 12] = [0x00; 12];
+
+
+
+// FROM LEFT
+// ADVTYPE      ;;      4 bits
+// RFU          ;;      2 bits
+// TxAdd        ;;      1 bit
+// RxAdd        ;;      1 bit
+// Length       ;;       6 bits
+// RFU          ;;      2 bits
+// AdvD         ;;      6 bytes
+// AdvData      ;;      31 bytes
+
+// Header (2 bytes) || Address (6 bytes) || Payload 31 bytes
+static mut PAYLOAD: [u8; 39] = [// ADV_IND, public addr  [HEADER]
+                                0x02,
+                                0xE,
+                                0x00,
+                                // Address          [ADV ADDRESS]
+                                0x90,
+                                0xD8,
+                                0x7A,
+                                0xBD,
+                                0xA3,
+                                0xED,
+                                // [LEN, AD-TYPE, LEN-1 bytes of data ...]
+                                // 0x09 - Local name
+                                // 0x54 0x6f 0x63 0x6b 0x4f 0x54 - TockOs
```

Comment:
Sure, we can remove it but it was more for convenience if a user only invokes "start_adv" then it will configured to "TockOS". 
BTW, the ["capsule"](https://github.com/frenicth/tock/blob/nrf51/ble_advertisement/chips/nrf51/src/ble_advertising_driver.rs) write over the entire payload once the offset is 0, either by start-up or invoking "clear_data"